Assignment6/RearrangingCars.py

```diff
@@ -17,30 +14,36 @@ def rearrange(self, result):
         """
         Rearrange cars in self._cars to get the result and print out each step.
         :param result: list(int)
-        :return: None
+        :return: None or raise ValueError
         """
-        cars = self._cars
-        moves = []
+        if len(result) != self.size():
+            raise ValueError

```

Comment:
Nice idea! No one told us to think about this, but it's a good case to consider and handle